draft-ietf-masque-connect-udp.md

```diff
@@ -292,6 +292,31 @@ reduce performance as both can sometimes independently retransmit the same
 data. To avoid this, HTTP/3 datagrams SHOULD be used.
 
 
+## Tunneling of ECN Marks
+
+CONNECT-UDP does not create an IP-in-IP tunnel, so the guidance in {{?RFC6040}}
+about transferring ECN marks between inner and outer IP headers does not apply.
+
+If a client or proxy with a connection containing a CONNECT-UDP stream disables
+congestion control in accordance with {{performance}}, it MUST NOT signal ECN
+support on that connection. That is, it MUST mark all IP headers with the
+Not-ECT codepoint. It MAY continue to report ECN feedback via ACK_ECN frames, as
+the peer may not have disabled congestion control.
+
+Note that CONNECT-UDP clients do not have the ability in this specification to
+control the ECN codepoints on datagrams the proxy sends to the server, nor can
+proxies communicate the markings of each incoming server packet to the client.
+
+Nevertheless, a CONNECT-UDP proxy MUST NOT copy the ECT codepoint from the
```

Comment:
This is hard to reason about, and I think I'm getting confused between this paragraph and the next. And the fact that a QUIC packet can carry multiple DATAGRAM frames makes things even harder. Could we instead say `A CONNECT-UDP proxy MUST ignore ECN bits in the IP header of UDP packets received from the server, and MUST NOT set the ECN bits on UDP packets it sends to the server.`